### PR TITLE
Rootless container, securityContext

### PR DESCRIPTION
**Pull request by C4tWithShell**
1. Change dockerfile to run it as rootless container
2. Add securityContext to the helm chart

### CODE REVIEW

**Comment by rndmh3ro** [comment]
Thanks for your contribution. We'll test it.

As for the code. We use [helmify](https://github.com/arttor/helmify/) to automatically create the helm-charts from the yaml-files in the deploy-folder.

helmify removes the if-condition for the suecrity-context:

```diff
96,99d95
<         {{- if .Values.securityContext.enabled }}
<         securityContext:
<           {{- toYaml .Values.operator.trivyDojoReportOperator.securityContext | nindent 10 }}
<         {{- end }}
107a104,105
>         securityContext: {{- toYaml .Values.operator.trivyDojoReportOperator.containerSecurityContext
>           | nindent 10 }}
64c64
<   namespace: '{{ .Release.Namespace }}'
---
>   namespace: '{{ .Release.Namespace }}'
\ No newline at end of file
8,9d7
< securityContext:
<   enabled: true
12,15c10,13
<     - name: metrics
<       port: 80
<       protocol: TCP
<       targetPort: metrics
---
>   - name: metrics
>     port: 80
>     protocol: TCP
>     targetPort: metrics
17a16,26
>     containerSecurityContext:
>       allowPrivilegeEscalation: false
>       capabilities:
>         drop:
>         - ALL
>       privileged: false
>       readOnlyRootFilesystem: true
>       runAsNonRoot: true
>       runAsUser: 1000
>       seccompProfile:
>         type: RuntimeDefault
25a35
>       defectDojoEnvName: Development
26a37
>       defectDojoEvalEnvName: "false"
29d39
<       defectDojoEvalEnvName: "false"
34d43
<       defectDojoEnvName: Development
39,49d47
<     securityContext:
<       runAsUser: 1000
<       runAsNonRoot: true
<       privileged: false
<       readOnlyRootFilesystem: true
<       allowPrivilegeEscalation: false
<       seccompProfile:
<         type: RuntimeDefault
<       capabilities:
<         drop:
<         - ALL
```

For me this is fine, I see no use in allowing it to be disabled. Are you fine with this, too?

**Comment by C4tWithShell** [comment]
Sure

File: Dockerfile [hunk 1]
```diff
@@ -11,8 +11,15 @@ RUN poetry install --no-ansi
 
 FROM python:3.12-slim@sha256:36d57d7f9948fefe7b6092cfe8567da368033e71ba281b11bb9eeffce3d45bc6
 
+RUN groupadd --gid 1000 app && \
```
Comment (szEvEz) [inline Dockerfile:14]: There is a permission issue when running the Container. I would like to do it like this:

```
FROM python:3.12@sha256:336461f63f4eb1100e178d5acbfea3d1a5b2a53dea88aa0f9b8482d4d02e981c as build

WORKDIR /app

RUN pip install --no-cache-dir poetry

COPY poetry.lock pyproject.toml /app/

RUN poetry config virtualenvs.in-project true
RUN poetry install --no-ansi

FROM python:3.12-slim@sha256:36d57d7f9948fefe7b6092cfe8567da368033e71ba281b11bb9eeffce3d45bc6

RUN groupadd --gid 1000 app && \
    useradd --gid 1000 --uid 1000 app

COPY --from=build /app /app

COPY src/* /app/

RUN chown -R app:app /app

USER app

WORKDIR /app

CMD ["/app/.venv/bin/kopf", "run", "--liveness=http://0.0.0.0:8080/healthz", "/app/handlers.py", "--all-namespaces"]
```

File: deploy/deployment.yaml [hunk 2]
```diff
@@ -17,9 +17,26 @@ spec:
...
+            runAsUser: 1000
+            runAsNonRoot: true
+            privileged: false
+            readOnlyRootFilesystem: true
+            allowPrivilegeEscalation: false
+            seccompProfile:
+              type: RuntimeDefault
+            capabilities:
+              drop:
+              - ALL
```
Comment (szEvEz) [inline deploy/deployment.yaml:39]:
```suggestion
      securityContext:
          runAsNonRoot: true
          fsGroupChangePolicy: Always
      containers:
        - name: trivy-dojo-report-operator
          image: ghcr.io/telekom-mms/docker-trivy-dojo-operator:0.5.7
          securityContext:
            runAsUser: 1000
            fsGroup: 1001
            privileged: false
            readOnlyRootFilesystem: true
            allowPrivilegeEscalation: false
            seccompProfile:
              type: RuntimeDefault
            capabilities:
              drop:
              - ALL
```

Comment (szEvEz) [reply deploy/deployment.yaml:39]: podSecurityContext is not a valid field in the deployment spec. I also moved some context between pod and container level

File: charts/values.yaml [hunk 3]
```diff
@@ -1,18 +1,30 @@
...
+    containerSecurityContext:
+      allowPrivilegeEscalation: false
+      capabilities:
+        drop:
+        - ALL
+      privileged: false
+      readOnlyRootFilesystem: true
+      runAsGroup: 1000
+      fsGroup: 1000
+      runAsUser: 1000
```
Comment (szEvEz) [inline charts/values.yaml:25]:
```suggestion
      allowPrivilegeEscalation: false
      capabilities:
        drop:
        - ALL
      privileged: false
      readOnlyRootFilesystem: true
      runAsGroup: 1000
      runAsUser: 1000
```

File: charts/values.yaml [hunk 4]
```diff
@@ -38,3 +50,6 @@ operator:
       repository: ghcr.io/telekom-mms/docker-trivy-dojo-operator
       tag: 0.5.7
   type: ClusterIP
+  podSecurityContext:
+    runAsNonRoot: true
+    fsGroupChangePolicy: Always
```
Comment (szEvEz) [inline charts/values.yaml:55]:
```suggestion
  podSecurityContext:
    runAsNonRoot: true
    fsGroupChangePolicy: Always
    fsGroup: 1000
```

File: deploy/deployment.yaml [hunk 5]
```diff
@@ -17,9 +17,24 @@ spec:
         application: trivy-dojo-report-operator
     spec:
       serviceAccountName: trivy-dojo-report-account
+      securityContext:
+        runAsNonRoot: true
+        fsGroupChangePolicy: Always
```
Comment (szEvEz) [inline deploy/deployment.yaml:22]:
```suggestion
      securityContext:
        runAsNonRoot: true
        fsGroupChangePolicy: Always
        fsGroup: 1000
```

File: deploy/deployment.yaml [hunk 6]
```diff
@@ -17,9 +17,24 @@ spec:
...
       containers:
         - name: trivy-dojo-report-operator
           image: ghcr.io/telekom-mms/docker-trivy-dojo-operator:0.5.7
+          securityContext:
+            runAsUser: 1000
+            runAsGroup: 1000
+            fsGroup: 1000
+            privileged: false
```
Comment (szEvEz) [inline deploy/deployment.yaml:30]:
```suggestion
          securityContext:
            runAsUser: 1000
            runAsGroup: 1000
            privileged: false
```

**Review by szEvEz — COMMENTED** [review]
Sorry, last modifications. fsgroup is only a pod securitycontext parameter, not a container level one.

**Review by szEvEz — APPROVED** [review]
LGTM!

**Pull request merged**